5.OAM_KPT_Demo/README.md

```diff
@@ -0,0 +1,281 @@
+              value: ""
+  parameters:
+    - name: instance-name
+      required: true
+      fieldPaths:
+        - metadata.name
+    - name: image
+      fieldPaths:
+        - spec.containers[0].image
+```
+
+The App Developer left two parameters which are `instance-name` and `image`.
+
+Then in AppConfig, App Operator could overwrite this parameter like below.
+
+```yaml
+apiVersion: core.oam.dev/v1alpha2
+kind: ApplicationConfiguration
+metadata:
+  name: example-appconfig
+spec:
+  components:
+    - componentName: example-component
+      parameterValues:
+        - name: instance-name
+          value: example-appconfig-workload
+        - name: image
+          value: nginx:latest
+      traits:
+        - trait:
+            apiVersion: core.oam.dev/v1alpha2
+            kind: ManualScalerTrait
+            metadata:
+              name:  example-appconfig-trait
+            spec:
+              replicaCount: 3
+```
+
+With kpt setters, this feature become more clear and useful. And you don't need to declare parameters in OAM yamls. Let's go through with it.
+
+#### Create setter by App Developer
+
+Let's say the developer need to claim two fields as "configurable" for his application, he can add two kpt setters here:
+
+```shell
+$ kpt cfg create-setter sampleapp/ instance-name example-component --field "metadata.name" --description "use to set an instance name" --set-by "sampleapp developer"
+```
+
+```shell
+$ kpt cfg create-setter sampleapp/ image nginx:1.16.1 --field "image" --description "use to set image for component" --set-by "sampleapp developer"
+```
+
+Then the app operator could see which parameters are available in this component like below:
+
+```shell
+$ kpt cfg list-setters sampleapp/
+  NAME              VALUE               SET BY                   DESCRIPTION             COUNT
+  image           nginx:1.16.1        sampleapp developer   use to set image for component   1
+  instance-name   example-component   sampleapp developer   use to set an instance name      1
+```
+
+It's very clear and easy to understand.
+
+#### Set Value by App Operator
+
+Then the application operator could set `instance-name` with a new name like this:
+
+```shell
+$ kpt cfg set sampleapp/ instance-name test-component
+set 1 fields
+```
+
+Check the component and you will find the instane name has been changed.
+
+```shell
+$ cat sampleapp/component.yaml
+apiVersion: core.oam.dev/v1alpha2
+kind: Component
+metadata:
+  name: test-component # {"$ref":"#/definitions/io.k8s.cli.setters.instance-name"}
+spec:
+  parameters:
+  - name: instance-name
+    fieldPaths:
+    - metadata.name
+...
+```
+
+### App Overview
+
+With kpt, you could see an overview of your App.
+
+```shell
+$ kpt cfg count sampleapp
+ApplicationConfiguration: 1
+Component: 1
+TraitDefinition: 1
+WorkloadDefinition: 1
+```
+
+So in the sampleapp, we have one ApplicationConfiguration, Component, TraitDefinition and WorkloadDefinition.
```

Comment:
It's better to make XxxDefinitions as part of Crossplane installation, ref: https://github.com/oam-dev/crossplane-oam-sample/issues/1